internals_fsutils.go

```diff
@@ -382,53 +386,51 @@ func createZip(archiveName string, files map[string][]byte) error {
 		if err != nil {
 			return err
 		}
-		_, err = f.Write([]byte(fcont))
+		_, err = io.Copy(f, fcont)
 		if err != nil {
 			return err
 		}
 	}
 
-	// Make sure to check the error on Close.
-	err := w.Close()
-	if err != nil {
-		return err
-	}
-
-	err = ioutil.WriteFile(archiveName, buf.Bytes(), defaultFilePermissions)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	// The Close call writes the checksum, so return the error
```

Comment:
Thanks for that snippet, quite helpful. I'm not familiar with Go's best practices in this area, so my comment was just to explain what the original author was doing -- but agree that improving it while we're here is desirable. In the snippet you gave, when `err` is set in the anonymous func, how is that error ever returned to the caller?